### PR TITLE
MM-20588: Migrate 'components/setting_item_min' module and associated tests to TypeScript

### PR DESCRIPTION
**Pull request by allenlai18**
#### Ticket Link

https://github.com/mattermost/mattermost-server/issues/13484
https://mattermost.atlassian.net/browse/MM-20588

### CODE REVIEW

**Comment by cpoile** [comment]
Hi @allenlai18, here are the changes that fix the type errors. Would love @saturninoabril 's opinion if this is the right way to go.

```
Index: components/setting_item_min/setting_item_min.test.tsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- components/setting_item_min/setting_item_min.test.tsx	(date 1578854242000)
+++ components/setting_item_min/setting_item_min.test.tsx	(date 1579289055000)
@@ -38,11 +38,11 @@
     test('should have called updateSection on handleUpdateSection with section', () => {
         const updateSection = jest.fn();
         const props = {...baseProps, updateSection};
-        const wrapper: SettingItemMin = shallow(
+        const wrapper = shallow<SettingItemMin>(
             <SettingItemMin {...props}/>
         );
 
-        wrapper.instance().handleUpdateSection({preventDefault: jest.fn()});
+        wrapper.instance().handleUpdateSection({preventDefault: jest.fn()} as any);
         expect(updateSection).toHaveBeenCalled();
         expect(updateSection).toHaveBeenCalledWith('section');
     });
@@ -50,11 +50,11 @@
     test('should have called updateSection on handleUpdateSection with empty string', () => {
         const updateSection = jest.fn();
         const props = {...baseProps, updateSection, section: ''};
-        const wrapper = shallow(
+        const wrapper = shallow<SettingItemMin>(
             <SettingItemMin {...props}/>
         );
 
-        wrapper.instance().handleUpdateSection({preventDefault: jest.fn()});
+        wrapper.instance().handleUpdateSection({preventDefault: jest.fn()} as any);
         expect(updateSection).toHaveBeenCalled();
         expect(updateSection).toHaveBeenCalledWith('');
     });
Index: components/setting_item_min/setting_item_min.tsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- components/setting_item_min/setting_item_min.tsx	(date 1578854242000)
+++ components/setting_item_min/setting_item_min.tsx	(date 1579289055000)
@@ -61,7 +61,7 @@
         this.edit = node;
     }
 
-    private handleUpdateSection = (e: React.MouseEvent<HTMLElement>) => {
+    handleUpdateSection = (e: React.MouseEvent<HTMLElement>) => {
         e.preventDefault();
         this.props.actions.updateActiveSection(this.props.section);
         this.props.updateSection(this.props.section);
```

**Comment by allenlai18** [comment]
@cpoile   thank you for creating the .patch .   That is a interesting way of making the types work. didn't know you can construct a shallow object like that

File: components/setting_item_min/setting_item_min.tsx [hunk 1]
```diff
@@ -1,79 +1,73 @@
...
-         */
-        actions: PropTypes.shape({
-
-            /**
-             * Update the active section for focusing
-             */
-            updateActiveSection: PropTypes.func.isRequired,
-        }).isRequired,
-    };
+interface SettingItemMinProps {
```
Comment (saturninoabril) [inline components/setting_item_min/setting_item_min.tsx]: I suggests to name it as `Props` only unless there's more than once component in a file that needs to distinguish props. 
```suggestion
interface Props {
```

File: components/setting_item_min/setting_item_min.tsx [hunk 2]
```diff
@@ -1,79 +1,73 @@
...
+    previousActiveSection?: string;
+
+    /**
+     * Actions
+     * Update the active section for focusing
+     */
+    actions: {updateActiveSection: (newActiveSection: string) => {type: string; data: string}};
+}
+
+export default class SettingItemMin extends React.PureComponent<SettingItemMinProps, {}> {
```
Comment (saturninoabril) [inline components/setting_item_min/setting_item_min.tsx]: Please remove empty object (state) if not to be used by this component.
```suggestion
export default class SettingItemMin extends React.PureComponent<Props> {
```

**Comment by allenlai18** [comment]
@cpoile  ready for review, when you get a chance

**Comment by cpoile** [comment]
Thanks for the contribution @allenlai18, looks great!

**Pull request merged**